### ISSUE TITLE
A few new ideas for the contribution! 

### ISSUE THREAD

**Issue by Halphas-Kreuz** [davidstone/technical-machine#114]
Hallo David! 

I would like to build an AI for pokemon battle ,and then I found your project! 
As I am new in C++ and not exactly understand how your program works, I have a few ideas. 

1. About Predicting the items 
        if the weather (sandstorm) lasts longer than 5 Turns : Sandstone (other weather stone can be detected in the same way)
        if the opponent has damaged your team more than (Opponents Pokemon attack, , full IV, full EV, max Random ) *1.3 = life orb, more than 1.5* then it must be choice glass . it would not take too long to calculate. (because that is how human understand the way too high damage) 
        and then, about team prediction : garchomp Max ATK SPD and garchomp max DEF HP are almost two species ! we may use the existed data to make a correlation ,see which type garchomp  is now more popular and while type is strong correlated with which other pokemon.

**Comment by davidstone** [comment]
Good ideas! The current usage stats are hard to parse, buggy, and leave out some important information. There has been a plan to update them for a while, but I might just have to write my own usage stat parser and contribute that to Smogon so that I can get the stats I need to be able to start working on some of the stuff like this. Obviously the "detect items from effects without messages" thing doesn't rely on that necessarily, but my thinking is that they would be connected because I could set the probability of having a certain item to 0 when I can rule it out.

Before I can do this effectively, however, I need to make another change I've been planning for a while. Right now, I have a single type, `Team` (technically it's `Team<generation>`, but that's not important for this). What I need to do is create two types: `Team` and `SeenTeam` (or maybe `FoeTeam`). `SeenTeam` can store as much information as it needs and it doesn't need to worry so much about efficiency, and then when I evaluate a position I predict a `Team` from a `SeenTeam` and use that. This gives me a place to put all the metadata I need for a prediction without slowing down the evaluation by making a `Team` larger for information that's not needed during evaluation.
